Src/Support/Google.Apis.Core/GoogleApiException.cs

```diff
@@ -39,24 +40,66 @@ public GoogleApiException(string serviceName, string message, Exception inner)
         {
             serviceName.ThrowIfNull("serviceName");
             this.serviceName = serviceName;
+            this.message = message;
         }
 
         /// <summary>Creates an API Service exception.</summary>
-        public GoogleApiException(string serviceName, string message) : this(serviceName, message, null) { }
+        public GoogleApiException(string serviceName, string message)
+            : this(serviceName, message, null)
+        { }
+
+        /// <summary>Creates an API Service exception.</summary>
+        public GoogleApiException(string serviceName)
+            : this(serviceName, null, null)
+        { }
 
         /// <summary>The Error which was returned from the server, or <c>null</c> if unavailable.</summary>
         public RequestError Error { get; set; }
 
+        private string ErrorMessage =>
+            Error?.Message ?? "No error message was specified.";
+
+        private string ErrorContentMessage =>
+            Error is null || Error.IsOnlyRawContent
+            ? "No JSON error details were specified."
+            : $"{Error}";
+
+        private string RawContentMessage =>
+            Error?.ErrorResponseContent is null
+            ? "No error details were specified."
+            : Error.ErrorResponseContent.Trim() == ""
+            ? $"Raw error details are empty or white spaces only."
+            : $"Raw error details are: {Error.ErrorResponseContent}";
+
         /// <summary>The HTTP status code which was returned along with this error, or 0 if unavailable.</summary>
         public HttpStatusCode HttpStatusCode { get; set; }
 
+        private string HttpStatusCodeMessage =>
+            HttpStatusCode > 0
+            ? $"HttpStatusCode is {HttpStatusCode}."
+            : "No HttpStatusCode was specified.";
+
+        private string ServiceNameMessage => $"The service {serviceName} has thrown an exception.";
+
+        private string DefaultMessage =>
+            $"{ServiceNameMessage} {HttpStatusCodeMessage} {ErrorMessage}";
+
+        /// <inheritdoc/>
+        public override string Message =>
+            // We just override the default message which is very generic in Exception
+            message is null ? DefaultMessage : base.Message;
+
         /// <summary>
         /// Returns a summary of this exception.
         /// </summary>
         /// <returns>A summary of this exception.</returns>
-        public override string ToString()
-        {
-            return string.Format("The service {1} has thrown an exception: {0}", base.ToString(), serviceName);
-        }
+        public override string ToString() =>
```

Comment:
Would this include the entirety of the JSON? I'm slightly nervous about that - it becomes a really big change in some cases.